connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/internal/UplynkAdIntegration.kt

```diff
@@ -38,30 +41,37 @@ internal class UplynkAdIntegration(
         val uplynkSource = source.sources.singleOrNull { it.ssai is UplynkSsaiDescription }
         val ssaiDescription = uplynkSource?.ssai as? UplynkSsaiDescription ?: return source
 
-        val response = uplynkDescriptionConverter
-            .buildPreplayUrl(ssaiDescription)
-            .let { uplynkApi.preplay(it) }
-            .also {
-                try {
-                    val response = it.parseExternalResponse()
-                    eventDispatcher.dispatchPreplayEvents(response)
-                    eventDispatcher.dispatchAdBreaksUpdatedEvents(response.ads)
-                    adScheduler = UplynkAdScheduler(response.ads.breaks, AdHandler(controller), eventDispatcher)
-                } catch (e: Exception) {
-                    eventDispatcher.dispatchPreplayFailure(e)
-                    controller.error(e)
-                }
-            }
+        val preplayUrl = uplynkDescriptionConverter.buildPreplayUrl(ssaiDescription)
+        val internalResponse = uplynkApi.preplay(preplayUrl)
+        val minimalResponse = internalResponse.parseMinimalResponse()
 
-        val internalResponse = response.parseMinimalResponse()
+        var newUplynkSource = uplynkSource.replaceSrc(minimalResponse.playURL)
+        minimalResponse.drm?.let { drm ->
+            val drmBuilder = DRMConfiguration.Builder()
+            drm.widevineLicenseURL?.let { drmBuilder.widevine(KeySystemConfiguration.Builder(it).build()) }
+            drm.playreadyLicenseURL?.let { drmBuilder.playready(KeySystemConfiguration.Builder(it).build()) }
+            drm.fairplayCertificateURL?.let { drmBuilder.fairplay(FairPlayKeySystemConfiguration.Builder("", it).build()) }
```

Comment:
Our `FairPlayKeySystemConfiguration.Builder` expects both non-null `licenseAcquisitionURL` & `certificateURL`.
Meanwhile for Uplynk the `licenseAcquisitionURL` would be retrieved from the EXT-X-KEY in the HLS manifest.

In theory at least for now, this shouldn't cause issues since we won't be playing FairPlay streams on Android. In the future we could make the `licenseAcquisitionURL` optional in the SDK.

Reply:
Yeah, this would only be relevant if:
1. You start casting from an Android sender
2. An iOS sender joins the same Chromecast session
3. The iOS sender leaves the session and starts playing the received source locally

Let's just not bother with that. 😅

Reply:
Currently if we have a source that has fairplay drm protection we cannot play it on Android THEOplayer. Event if the playback is not protected but we just fill in fairplay configuration THEOplayer does not play this source without any error.

I'm not sure if it is a bug or desired behaviour, actually, but it works like this now. I think it's even less chances that this use case would happen. 

Should we even keep this line?
`
 drm.fairplayCertificateURL?.let { drmBuilder.fairplay(FairPlayKeySystemConfiguration.Builder("", it).build()) }`

Reply:
If it doesn't work at all with a Fairplay configuration, then yes, comment or remove it.